### ISSUE TITLE
Do you accept tf.keras.callbacks?

### ISSUE THREAD

**Issue by beangoben** [tensorflow/addons#58]
Seeing as keras is going to become a first citizen inside of tensorflow, are you also accepting submission for callbacks? Are there guidelines for this (writing tests/docs)?.

**Comment by seanpmorgan** [comment]
My first thought is that it seems like it would be in scope as there is a pre-defined structure. I'll add it to the [monthly meeting this Friday](https://docs.google.com/document/d/1kxg5xIHWLY7EMdOJCdSGgaPu27a9YKpupUz2VTXqTJg/edit#) where we can discuss adding this as a new standardized API. 

I'll update this issue after the meeting... but can you give some example callbacks you'd like to see added and/or would be willing to contribute?

**Comment by seanpmorgan** [comment]
Also, just a reminder... the monthly meeting is at 1pm PST (join our [google group](https://groups.google.com/a/tensorflow.org/forum/#!forum/addons) and you'll get calendar invites)  and we encourage all to come!

**Comment by beangoben** [comment]
Would have to check, but potentially some callbacks would be:
* LearningRateScheduler compatible functions like SGDR (https://www.tensorflow.org/api_docs/python/tf/train/cosine_decay_restarts).
* In-memory best weights restore (like ModelCheckpoint but in-memory).
* Evaluation of metrics (r^22,auroc,f1,etc.) on external datasets (test set) each epoch, batch. 
* Saving 'snapshots' of your embeddings during training.

**Comment by facaiy** [comment]
Do you mean to implement them as subclass of [tf.keras.callbacks.Callback](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback)?

**Comment by beangoben** [comment]
Yes, with tests..but that all depends on the guidelines that will be recommended/set.

**Comment by seanpmorgan** [comment]
Short answer to this issue is Yes. See #61
